### DISCUSSION TITLE
Guideline for Component Positioning in Timeline (use cases: add delay/gap between sequences, Audio positioning,...)

### DISCUSSION THREAD

**Discussion by mnlbox** [remotion-dev/remotion#3501] [Q&A]
Hi,

I have a composition contains multiple sequences. Now I need to add 5 frame gap between each sequences. I know I can do it with the combination of `durationInFrame` or `from` props but for that I need to do lot's of calculation for each sequence.
Is there any easier way to add this 5 frame gap between each sequences?
Is there any helper or component available for this type of use cases to position each sequence in the video timeline?

**Comment by mnlbox** [comment]
I think I need to learn how to position each component in the timeline in general then I can solve the `gap` issue that I have and I mentioned in the previous comment too. Because now I have same problem again about positioning `<Audio>` components. I have a peace of code like this:
```
{videoData.rows.voices.map((voice, index) => (
    <Audio
        key={index}
        src={staticFile(voice.path)}
        autoPlay={currentVoiceIndex === index}
    />
))}
```
<img width="1055" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/5522842c-9ba8-4a09-b3ce-92de2686c836">

But in the timeline all of them started at same position, while I need to play them in sequence. Do I need to create separate `sequence` component for each one of my `<Audio>` components here?

**Comment by mnlbox** [comment]
OK after doing more research I found `Series` components [here](https://www.remotion.dev/docs/series) from Remotion package useful for my use-case, I guess I can use it for positioning and add gap as it has an `offset` property.

But It seems `durationInFrames` is mandatory for `<Series.Sequence>` component. So I have to calculate the `<Audio>` duration and pass it to the `<Series.Sequence>` component. I tried something like this:
```
<Series>
    {videoData.rows.voices.map(
        async (voice, index) => {
            const audioData = await getAudioData(staticFile(voice.path));
            const durationInSeconds = audioData.durationInSeconds;
            const durationInFrame = Math.round(durationInSeconds * fps);

            return (
                <Series.Sequence durationInFrames={durationInFrame} key={index}>
                    <Audio src={staticFile(voice.path)} />
                </Series.Sequence>
            );
        },
    )}
</Series>
```

But as `getAudioData` is an async function, it's not working. Trying to solve this issue now...

**Comment by mnlbox** [comment]
OK I managed to fix that using `<Series.Sequence>` the only problem is that for each sequence one track created so in general it looks like a waterfall:
<img width="1169" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/ed7a53db-aca7-4f38-8686-f98387a399a7">

So I don't know if `<Series.Sequence>` is the right way for my use-case or not?
Maybe related: https://github.com/orgs/remotion-dev/discussions/3111

**Reply by JonnyBurger** [reply]
You can indeed use an empty `<Series.Sequence>` to create gaps.

Or actually leverage the `offset` property of it (with a negative value I think should work)!

There is also `showInTimeline={false}` that you can add to any Sequence, and should work as well for `<Series.Sequence>` I believe.

**Comment by mnlbox** [comment]
Thanks @JonnyBurger  For your reply.

I have the timeline bellow:
<img width="858" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/a2c7a8bf-7d64-4486-a227-6b04b961242f">

I tried `showInTimeline={false}` prop as bellow:

```
<Sequence
    from={title.durationInFrames}
    durationInFrames={intro.durationInFrames}
    showInTimeline={false}>
        <Audio src={staticFile("/intro.mp3")} />
</Sequence>
```

But it still shows the Audio there, This is my timeline after `showInTimeline={false}`:
<img width="858" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/1609e990-9033-4b6e-8d4a-8f868b41b521">

Is it normal or it's a bug? 🤔 

I still think having an ability to have multiple sequences from same `<Series>` component in a single track can be a nice feature, something like this:
<img width="1107" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/39165b34-2ce7-4693-aa2b-bc9d4370c08e">

**Reply by JonnyBurger** [reply]
We'll add a new `showInTimeline` props to all media tags: https://github.com/remotion-dev/remotion/pull/3550

Remotion's timeline works more like the one in After Effects or Photoshop, it is not conceptually the same as a timeline in a normal video editing program.
